lib/pdk/config.rb

```diff
@@ -43,7 +49,26 @@ def user
     # @param filter [String] Only resolve setting names which match the filter. See PDK::Config::Namespace.be_resolved? for matching rules
     # @return [Hash{String => Object}] All resolved settings for example {'user.module_defaults.author' => 'johndoe'}
     def resolve(filter = nil)
-      user.resolve(filter)
+      self.system.resolve(filter).merge(user.resolve(filter)) # rubocop:disable Style/RedundantSelf Use an explicit call so as to not confuse with Kernel.system
```

Comment:
Hmmm, yeah how to we feel about shadowing `Kernel.system` in this class? Is the brevity of `.user` and `.system` worth the potential collision? Should we just rename the methods to `.user_config` and `.system_config` even though that is slightly redundant?

Reply:
Yeah I thought long and hard about this too.  I eventually conceded that the brevity is useful because, that method will _normally_ be called via 'PDK,config.system`.  The shadowing is only really a problem as an implementation detail within `Config` class.

In fact, ideally `PDK,config.system` should never actually be called.  I should really mark that as `@api private` because once `Config.get` and `Config.set` is complete ALL access to `.system` will be internal to the object and won't need to be on the public API at all.

Reply:
So in a round_about way I'm saying....

I should change it to `user_config` and `system_config` with the intention to mark those methods as private ASAP.

Reply:
Done.

Reply:
Not marking as resolved for the moment, but has been implemented